### PR TITLE
Add solution for lab06

### PR DESCRIPTION
**Pull request by lucigrigo**

### CODE REVIEW

File: labs/lab06/skel/lin/3-efence/bug.c [hunk 1]
```diff
@@ -14,18 +14,18 @@
 
 int main(void)
 {
-	int i;
-	int *v = calloc(10, sizeof(int));
-	int *w = calloc(10, sizeof(int));
+    int i;
+    int *v = calloc(10, sizeof(int));
+    int *w = calloc(10, sizeof(int));
```
Comment (teodutu) [inline labs/lab06/skel/lin/3-efence/bug.c]:
```suggestion
    int *v = calloc(10, sizeof(*v));
    int *w = calloc(10, sizeof(*w));
```

File: labs/lab06/skel/lin/7-paging/paging.c [hunk 2]
```diff
@@ -18,60 +18,59 @@
...
-	lock_memory(data, DATA_SIZE);
-	wait_for_input("memory locked ");
+    lock_memory(data, DATA_SIZE);
+    wait_for_input("memory locked ");
 
-	/*  perform real-time data changes */
-	memcpy(data, text, sizeof(text));
-	printf("data=%s\n", data);
-	wait_for_input("accessed memory");
+    /*  perform real-time data changes */
```
Comment (teodutu) [inline labs/lab06/skel/lin/7-paging/paging.c]:
```suggestion
    /* Perform real-time data changes */
```

File: labs/lab06/skel/win/4-prot/libvm.c [hunk 3]
```diff
@@ -23,67 +23,67 @@
...
-	/* TODO 2 - get the page number which caused the page fault */
+    /* TODO - get the page number which caused the page fault */
 
+    /* TODO - test if page is one of our own */
 
-	/* TODO 2 - test if page is one of our own */
+    /* TODO - increase protection for that page */
 
-
-	/* TODO 2 - increase protection for that page */
```
Comment (teodutu) [inline labs/lab06/skel/win/4-prot/libvm.c]: Keep TODO numbers. They're useful for the students.

**Review by teodutu — CHANGES_REQUESTED** [review]
I left some required changes as inline suggestions. The rest of the suggestions apply globally:

- Remove the `.DS_Store` and `labs/.DS_Store` files
- Revert the indentation from 4 spaces to tabs.
- Use the following style for multiline comments:
```
/**
 *
 */
```
- Use the Linux Kernel coding style for multiline `if`s, `for`s etc:
```c
if (cond) {
	/* stuff */
}
```
- In general, run `checkpatch.pl` and fix the coding style. Ignore the lack of an SPDX license and the CRLF line endings in Windows files.
- Start each comment with the capital letter, including the text after `/* TODO - ... */`.
- Use C89 comments (`/* ... */`), not C99 comments (`// ...`).
- Remove `labs/lab06/sol/.gitignore`. It's empty.
- Remove the empty line after the block comments at the beginning of `sol` files.

File: labs/lab06/skel/win/2-dyn/dyn.c [hunk 4]
```diff
@@ -30,26 +29,26 @@ int (*dyncode)(int);
  * (3) returns that value
  */
 unsigned char code[] = {
-	0x8B, 0x44, 0x24, 0x04,  /* mov eax, [esp+4] (1) */
-	0x40,			 /* inc eax          (2) */
-	0xC3			 /* ret              (3) */
+    0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
+    0x40,                   /* inc eax          (2) */
+    0xC3                    /* ret              (3) */
```
Comment (teodutu) [inline labs/lab06/skel/win/2-dyn/dyn.c]: Use 8-spaces wide tabs when indenting these.

File: labs/lab06/skel/lin/4-cp/mycp.c [hunk 5]
```diff
@@ -17,34 +17,34 @@
...
 
-    /* Open/create the output file */
-    fdout = open(argv[2], O_RDWR | O_CREAT | O_TRUNC, 0644);
-    DIE(fdout == -1, "open fdout");
+	/* Open/create the output file */
+	fdout = open(argv[2], O_RDWR | O_CREAT | O_TRUNC, 0644);
+	DIE(fdout == -1, "open fdout");
 
-    /* TODO - truncate the output file to the input file size */
+	/* TODO - truncate the output file to the input file size */
```
Comment (teodutu) [inline labs/lab06/skel/lin/4-cp/mycp.c]:
```suggestion
	/* TODO - Truncate the output file to the input file size */
```

File: labs/lab06/skel/lin/4-cp/mycp.c [hunk 6]
```diff
@@ -17,34 +17,34 @@
...
 
-    /* TODO - mmap the input and output file */
+	/* TODO - Mmap the input and output file */
```
Comment (teodutu) [inline labs/lab06/skel/lin/4-cp/mycp.c]:
```suggestion
	/* TODO - mmap the input and output file */
```

File: labs/lab06/skel/win/2-dyn/dyn.c [hunk 7]
```diff
@@ -23,32 +23,34 @@ static int pageSize = 0x1000;
...
  * (2) increments that value
  * (3) returns that value
  */
 unsigned char code[] = {
-    0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
-    0x40,                   /* inc eax          (2) */
-    0xC3                    /* ret              (3) */
+	0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
+	0x40,                   /* inc eax          (2) */
+	0xC3                    /* ret              (3) */
```
Comment (teodutu) [inline labs/lab06/skel/win/2-dyn/dyn.c]: Indent the comments using tabs, not spaces.

File: labs/lab06/sol/win/2-dyn/dyn.c [hunk 8]
```diff
@@ -23,32 +23,34 @@ static int pageSize = 0x1000;
...
  * (2) increments that value
  * (3) returns that value
  */
 unsigned char code[] = {
-    0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
-    0x40,                   /* inc eax          (2) */
-    0xC3                    /* ret              (3) */
+	0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
+	0x40,                   /* inc eax          (2) */
+	0xC3                    /* ret              (3) */
```
Comment (teodutu) [inline labs/lab06/sol/win/2-dyn/dyn.c]: Align the comments using tabs, not spaces.

**Review by teodutu — CHANGES_REQUESTED** [review]
The code looks about right. You still need to:
- Align the `\`s within `DIE` macros using tabs instead of spaces in all `utils.h` files.
- Remove `lab06/sol/.DS_Store`.

File: labs/lab06/skel/lin/5-prot/prot.c [hunk 9]
```diff
@@ -18,29 +18,32 @@
...
 static void segv_handler(int signum, siginfo_t *info, void *context)
 {
 	char *addr;
 	int rc;
 
-	/* Calling the old signal handler by default for TODO 1 */
-	/* Comment this line when solving TODO 2 */
+	/**
+	 * Calling the old signal handler by default for TODO 1
+	 * Comment this line when solving TODO 2
```
Comment (andreia-oca) [inline labs/lab06/skel/lin/5-prot/prot.c]:
```suggestion
	 * Comment the line below when solving TODO 2
```

File: labs/lab06/sol/lin/4-cp/mycp.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "utils.h"
+
+int main(int argc, char *argv[])
+{
+	int fdin, fdout, rc;
```
Comment (andreia-oca) [inline labs/lab06/sol/lin/4-cp/mycp.c]:
```suggestion
	int fdin;
	int fdout;
	int rc;
```

File: labs/lab06/sol/lin/4-cp/mycp.c [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+	char *src, *dst;
```
Comment (andreia-oca) [inline labs/lab06/sol/lin/4-cp/mycp.c]:
```suggestion
	char *src;
	char *dst;
```

File: labs/lab06/sol/lin/7-paging/paging.c [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+	printf(" * %s\n", msg);
+	printf(" -- Press ENTER to continue ...");
+	fflush(stdout);
+	fgets(buf, 32, stdin);
+}
+
+/* Lock memory interval [addr, addr + size - 1] */
+static void lock_memory(char *addr, size_t size)
+{
+	unsigned long page_offset, pagesize;
```
Comment (andreia-oca) [inline labs/lab06/sol/lin/7-paging/paging.c]:
```suggestion
	unsigned long page_offset;
	unsigned long pagesize;
```

File: labs/lab06/sol/lin/7-paging/paging.c [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+	/* TODO - Lock memory */
+	rc = mlock(addr, size);
+	DIE(rc == -1, "mlock");
+}
+
+/* Unlock memory interval [addr, addr + size - 1] */
+static void unlock_memory(char *addr, size_t size)
+{
+	unsigned long page_offset, pagesize;
```
Comment (andreia-oca) [inline labs/lab06/sol/lin/7-paging/paging.c]:
```suggestion
	unsigned long page_offset;
	unsigned long pagesize;
```

File: labs/lab06/sol/win/1-intro/intro.c [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+		0,
+		0);
+	DIE(p == NULL, "MapViewOfFile");
+
+	return p;
+}
+
+int main(void)
+{
+	DWORD rc, dwPageSize;
```
Comment (andreia-oca) [inline labs/lab06/sol/win/1-intro/intro.c]:
```suggestion
	DWORD rc;
	DWORD dwPageSize;
```

File: labs/lab06/sol/win/4-prot/libvm.c [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+{
+	int rc = RemoveVectoredExceptionHandler(access_violation_handler);
+
+	DIE(rc == FALSE, "RemoveVectoredExceptionHandler");
+}
+
+int main(void)
+{
+	BYTE ch;
+	DWORD old, rc;
```
Comment (andreia-oca) [inline labs/lab06/sol/win/4-prot/libvm.c]:
```suggestion
	DWORD old;
	DWORD rc;
```

File: labs/lab06/sol/win/5-my_efence/my_efence.c [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+	 */
+	dwOffest = dwPageSize - dwSize % dwPageSize;
+	return lpMem + dwOffest;
+}
+
+int main(void)
+{
+	int i;
+
+	/* int *v = malloc(10 * sizeof(int)); */
```
Comment (andreia-oca) [inline labs/lab06/sol/win/5-my_efence/my_efence.c]: Delete this comment.

**Review by andreia-oca — CHANGES_REQUESTED** [review]
Looks good to me. I added a few suggestions, mainly one-variable declaration per line.

File: labs/lab06/skel/win/2-dyn/dyn.c [hunk 17]
```diff
@@ -30,9 +30,9 @@ int (*dyncode)(int);
  * (3) returns that value
  */
 unsigned char code[] = {
-	0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
-	0x40,                   /* inc eax          (2) */
-	0xC3                    /* ret              (3) */
+	0x8B, 0x44, 0x24, 0x04,	/* mov eax, [esp+4]	(1) */
+	0x40,					/* inc eax			(2) */
+	0xC3					/* ret				(3) */
```
Comment (teodutu) [inline labs/lab06/skel/win/2-dyn/dyn.c]:
```suggestion
	0x8B, 0x44, 0x24, 0x04,	/* mov eax, [esp+4]	(1) */
	0x40,			/* inc eax		(2) */
	0xC3			/* ret			(3) */
```

File: labs/lab06/sol/win/2-dyn/dyn.c [hunk 18]
```diff
@@ -30,9 +30,9 @@ int (*dyncode)(int);
  * (3) returns that value
  */
 unsigned char code[] = {
-	0x8B, 0x44, 0x24, 0x04, /* mov eax, [esp+4] (1) */
-	0x40,                   /* inc eax          (2) */
-	0xC3                    /* ret              (3) */
+	0x8B, 0x44, 0x24, 0x04,	/* mov eax, [esp+4]	(1) */
+	0x40,					/* inc eax			(2) */
+	0xC3					/* ret				(3) */
```
Comment (teodutu) [inline labs/lab06/sol/win/2-dyn/dyn.c]:
```suggestion
	0x8B, 0x44, 0x24, 0x04,	/* mov eax, [esp+4]	(1) */
	0x40,			/* inc eax		(2) */
	0xC3			/* ret			(3) */

```

File: labs/lab06/skel/win/utils/utils.h [hunk 19]
```diff
@@ -1,23 +1,22 @@
...
+/* Useful macro for handling error codes */
 #define DIE(assertion, call_description)				\
 	do {								\
 		if (assertion) {					\
-			fprintf(stderr, "(%s, %s, %d): ",		\
-				__FILE__, __FUNCTION__, __LINE__);	\
-			PrintLastError(call_description);		\
+			fprintf(stderr, "(%s, %d): ",			\
+					__FILE__, __LINE__);		\
+			perror(call_description);			\
```
Comment (teodutu) [inline labs/lab06/skel/win/utils/utils.h]: Use `PrintLastError`. It's more specific on Windows.

File: labs/lab06/sol/win/utils/utils.h [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+#include <windows.h>
+#include <stdio.h>
+
+/* Useful macro for handling error codes */
+#define DIE(assertion, call_description)				\
+	do {								\
+		if (assertion) {					\
+			fprintf(stderr, "(%s, %d): ",			\
+					__FILE__, __LINE__);		\
+			perror(call_description);			\
```
Comment (teodutu) [inline labs/lab06/sol/win/utils/utils.h]: Use `PrintLastError`. It's more specific on Windows.

File: labs/lab06/skel/win/2-dyn/dyn.c [hunk 21]
```diff
@@ -24,25 +23,27 @@ static int pageSize = 0x1000;
...
-	/* static code will be copied to an area alocated with VirtualAlloc
+	/**
+	 * Static code will be copied to an area alocated with VirtualAlloc
 	 * - ergo any dynamically generated code could be treated this way
 	 */
 
-	dyncode = (int (*)(int)) VirtualAlloc(NULL, pageSize,
-		MEM_COMMIT, PAGE_EXECUTE_READWRITE);
+	dyncode = (int (*)(int))VirtualAlloc(NULL, pageSize,
+										 MEM_COMMIT, PAGE_EXECUTE_READWRITE);
```
Comment (teodutu) [inline labs/lab06/skel/win/2-dyn/dyn.c]:
```suggestion
	dyncode = (int (*)(int))VirtualAlloc(NULL, pageSize,
			MEM_COMMIT, PAGE_EXECUTE_READWRITE);
```

File: labs/lab06/skel/win/5-my_efence/my_efence.c [hunk 22]
```diff
@@ -31,37 +34,32 @@ static LPVOID MyMalloc(DWORD dwSize)
...
 	for (i = 0; i <= 10; i++) {
 		__try {
-			// Write to memory.
+			/* Write to memory. */
 			v[i] = i;
 		}
-		__except(GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ?
-			EXCEPTION_EXECUTE_HANDLER :
-			EXCEPTION_CONTINUE_SEARCH) {
+		__except (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ? EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH) {
```
Comment (teodutu) [inline labs/lab06/skel/win/5-my_efence/my_efence.c]: Break this into multiple lines so they don't exceed 80 characters.

File: labs/lab06/skel/lin/1-intro/intro.c [hunk 23]
```diff
@@ -38,12 +39,12 @@ int main(void)
...
 	wait_for_input("before mmap file use pmap to see file mapping");
 
-	/** First we map a file */
+	/* First we map a file */
 	fd = open("Makefile", O_RDWR);
 	DIE(fd == -1, "open");
 
 	p = mmap(NULL, page_size,
-			PROT_READ | PROT_WRITE, MAP_PRIVATE, fd, 0);
+			 PROT_READ | PROT_WRITE, MAP_PRIVATE, fd, 0);
```
Comment (teodutu) [inline labs/lab06/skel/lin/1-intro/intro.c]:
```suggestion
			PROT_READ | PROT_WRITE, MAP_PRIVATE, fd, 0);
```

File: labs/lab06/skel/lin/1-intro/intro.c [hunk 24]
```diff
@@ -54,19 +55,19 @@ int main(void)
 	rc = close(fd);
 	DIE(rc == -1, "close");
 
-	/** Second we map SHARED memory */
+	/* Second we map SHARED memory */
 	p = mmap(NULL, page_size, PROT_READ | PROT_WRITE,
-				 MAP_ANONYMOUS | MAP_SHARED, -1, 0);
+			 MAP_ANONYMOUS | MAP_SHARED, -1, 0);
```
Comment (teodutu) [inline labs/lab06/skel/lin/1-intro/intro.c]:
```suggestion
			MAP_ANONYMOUS | MAP_SHARED, -1, 0);
```

File: labs/lab06/skel/lin/1-intro/intro.c [hunk 25]
```diff
@@ -54,19 +55,19 @@ int main(void)
...
 
 	rc = munmap(p, page_size);
 	DIE(rc == -1, "munmap");
 	wait_for_input("after unmapping SHARED memory");
 
-	/** Thrid we map PRIVATE memory */
+	/* Third we map PRIVATE memory */
 	p = mmap(NULL, page_size, PROT_READ | PROT_WRITE,
-				 MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+			 MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
```
Comment (teodutu) [inline labs/lab06/skel/lin/1-intro/intro.c]:
```suggestion
			MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
```

File: labs/lab06/skel/lin/2-compare/mmap.c [hunk 26]
```diff
@@ -33,7 +34,7 @@ int main(void)
 	DIE(rc == -1, "ftruncate");
 
 	mem = mmap(0, N * sizeof(msg), PROT_READ | PROT_WRITE,
-		   MAP_SHARED, fd, 0);
+			   MAP_SHARED, fd, 0);
```
Comment (teodutu) [inline labs/lab06/skel/lin/2-compare/mmap.c]:
```suggestion
			MAP_SHARED, fd, 0);
```

File: labs/lab06/skel/lin/6-faults/fork-faults.c [hunk 27]
```diff
@@ -35,45 +36,45 @@ int main(void)
 	wait_for_input("beginning");
 
 	p = mmap(NULL, NUM_PAGES * page_size, PROT_READ | PROT_WRITE,
-			MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+			 MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
```
Comment (teodutu) [inline labs/lab06/skel/lin/6-faults/fork-faults.c]:
```suggestion
			MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
```

File: labs/lab06/skel/lin/8-hack/hack.c [hunk 28]
```diff
@@ -34,39 +34,42 @@ int foo(void)
...
+	/* Pid */
 	printf("process id %i\n", getpid());
 
 	printf("primul apel: foo intoarce %d\n", foo());
 
-	// schimbare protectie pagina
-	if (mprotect(addr_aligned, pagesize, PROT_READ|PROT_WRITE|PROT_EXEC)) {
+	/* Increase page protection */
+	if (mprotect(addr_aligned, pagesize, PROT_READ | PROT_WRITE | PROT_EXEC))
+	{
```
Comment (teodutu) [inline labs/lab06/skel/lin/8-hack/hack.c]:
```suggestion
	if (mprotect(addr_aligned, pagesize,
		PROT_READ | PROT_WRITE | PROT_EXEC)) {
```

File: labs/lab06/skel/lin/8-hack/hack.c [hunk 29]
```diff
@@ -34,39 +34,42 @@ int foo(void)
...
 		return 1;
 	}
 
-	// schimbare valoare de intors functie
+	/* Store function pointer */
 	f = (int (*)())foo;
 
 	while ((int)f < (int)page_end &&
-		((int)*f != 0xb8 && (char)*((char *)f + 1) != 0x0c))
+		   ((int)*f != 0xb8 && (char)*((char *)f + 1) != 0x0c))
```
Comment (teodutu) [inline labs/lab06/skel/lin/8-hack/hack.c]:
```suggestion
		((int)*f != 0xb8 && (char)*((char *)f + 1) != 0x0c))
```

File: labs/lab06/skel/win/1-intro/intro.c [hunk 30]
```diff
@@ -96,10 +97,10 @@ int main(void)
...
 
 	mem = VirtualAlloc(
-			NULL,			// System selects address
-			dwPageSize,		// Size of allocation
-			MEM_COMMIT,		// Allocate reserved pages
-			PAGE_READWRITE);	// Protection
+		NULL,            /* System selects address */
+		dwPageSize,      /* Size of allocation */
+		MEM_COMMIT,      /* Allocate reserved pages */
+		PAGE_READWRITE); /* Protection */
```
Comment (teodutu) [inline labs/lab06/skel/win/1-intro/intro.c]:
```suggestion
		NULL,			/* System selects address */
		dwPageSize,		/* Size of allocation */
		MEM_COMMIT,		/* Allocate reserved pages */
		PAGE_READWRITE);	/* Protection */
```

File: labs/lab06/sol/win/1-intro/intro.c [hunk 31]
```diff
@@ -0,0 +1,117 @@
...
+	rc = CloseHandle(hFile);
+	DIE(rc == FALSE, "CloseHandle");
+
+	wait_for_input("closing mapped file");
+
+	mem = VirtualAlloc(
+		NULL,            /* System selects address */
+		dwPageSize,      /* Size of allocation */
+		MEM_COMMIT,      /* Allocate reserved pages */
+		PAGE_READWRITE); /* Protection */
```
Comment (teodutu) [inline labs/lab06/sol/win/1-intro/intro.c]:
```suggestion
		NULL,			/* System selects address */
		dwPageSize,		/* Size of allocation */
		MEM_COMMIT,		/* Allocate reserved pages */
		PAGE_READWRITE);	/* Protection */
```

File: labs/lab06/sol/win/2-dyn/dyn.c [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+
+int main(void)
+{
+	/**
+	 * Static code will be copied to an area alocated with VirtualAlloc
+	 * - ergo any dynamically generated code could be treated this way
+	 */
+
+	dyncode = (int (*)(int))VirtualAlloc(NULL, pageSize,
+										 MEM_COMMIT, PAGE_EXECUTE_READWRITE);
```
Comment (teodutu) [inline labs/lab06/sol/win/2-dyn/dyn.c]:
```suggestion
	dyncode = (int (*)(int))VirtualAlloc(NULL, pageSize,
			MEM_COMMIT, PAGE_EXECUTE_READWRITE);
```

File: labs/lab06/sol/win/5-my_efence/my_efence.c [hunk 33]
```diff
@@ -0,0 +1,79 @@
...
+	lpMem = VirtualAlloc(NULL, dwPageSize * (dwNoPages + 1),
+						 MEM_COMMIT, PAGE_READWRITE);
+	DIE(lpMem == NULL, "VirtualAlloc");
+
+	/**
+	 * TODO - Change accces for the last page (guard page)
+	 * to PAGE_NOACCESS
+	 */
+	rc = VirtualProtect(lpMem + dwNoPages * dwPageSize, dwPageSize,
+						PAGE_NOACCESS, &oldProt);
```
Comment (teodutu) [inline labs/lab06/sol/win/5-my_efence/my_efence.c]:
```suggestion
	rc = VirtualProtect(lpMem + dwNoPages * dwPageSize, dwPageSize,
		PAGE_NOACCESS, &oldProt);
```

File: labs/lab06/sol/win/5-my_efence/my_efence.c [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+
+	GetSystemInfo(&siSysInfo);
+	dwPageSize = siSysInfo.dwPageSize;
+
+	/* Number of pages used to allocate a buffer of size dwSize */
+	dwNoPages = dwSize / dwPageSize + 1;
+
+	/* TODO - Alocate dwNoPages + 1 pages*/
+	lpMem = VirtualAlloc(NULL, dwPageSize * (dwNoPages + 1),
+						 MEM_COMMIT, PAGE_READWRITE);
```
Comment (teodutu) [inline labs/lab06/sol/win/5-my_efence/my_efence.c]:
```suggestion
	lpMem = VirtualAlloc(NULL, dwPageSize * (dwNoPages + 1),
		MEM_COMMIT, PAGE_READWRITE);
```

File: labs/lab06/sol/win/5-my_efence/my_efence.c [hunk 35]
```diff
@@ -0,0 +1,79 @@
...
+	int i;
+
+	int *v = MyMalloc(10 * sizeof(int));
+
+	for (i = 0; i <= 10; i++) {
+		__try {
+			/* Write to memory. */
+			v[i] = i;
+		}
+		__except (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ? EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH) {
```
Comment (teodutu) [inline labs/lab06/sol/win/5-my_efence/my_efence.c]: Break this into multiple lines so that they don't exceed 80 characters.

**Review by teodutu — CHANGES_REQUESTED** [review]
The number of tabs is a bit off at times. Other than that, it's looking good.

**Review by teodutu — APPROVED** [review]
👍

**Review by andreia-oca — APPROVED** [review]
Looks good to me.

**Pull request merged**